embassy-stm32/src/usart/buffered.rs

```diff
@@ -129,6 +125,10 @@ unsafe fn on_interrupt(r: Regs, state: &'static State) {
             });
         }
     }
+
+    // For others it is cleared above with `clear_interrupt_flags`.
+    #[cfg(any(usart_v1, usart_v2))]
+    sr(r).modify(|w| w.set_tc(false));
```

Comment:
This may not be necessary because the bit seems to be cleared by reading from the status register and writing to the data register. So maybe this can be moved back, and only `tcie` can be checked.